### PR TITLE
ESDT native payment factories

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Integrated sovereign account creator into run type components to be injected from `nodeRunner` and `sovereignNodeRunner`
  
## Proposed changes
- Account factory component is now created inside `runTypeComponents`
- Run type components now needs core components to call `Create`
- Fixed several random failing unit tests

## Testing procedure
- Local testnet

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: cmd/sovereignnode/sovereignNodeRunner.go [hunk 1]
```diff
@@ -313,6 +307,12 @@ func (snr *sovereignNodeRunner) executeOneComponentCreationCycle(
 		return true, err
 	}
 
+	log.Debug("creating runType components")
+	managedRunTypeComponents, err := snr.CreateManagedRunTypeComponents(managedCoreComponents, *configs.SovereignExtraConfig)
```
Comment (axenteoctavian) [inline cmd/sovereignnode/sovereignNodeRunner.go:311]: We need to check the `NativeESDT` validity before this

File: process/block/metablock_test.go [hunk 2]
```diff
@@ -3,13 +3,15 @@ package block_test
...
 	"errors"
-	"github.com/multiversx/mx-chain-go/testscommon/sovereign"
 	"math/big"
 	"reflect"
 	"sync"
 	"testing"
 	"time"
 
+	"github.com/multiversx/mx-chain-go/testscommon/components"
+	"github.com/multiversx/mx-chain-go/testscommon/sovereign"
```
Comment (sasurobert) [inline process/block/metablock_test.go:13]: goimports

File: testscommon/mainFactoryMocks/runTypeComponentStub.go [hunk 3]
```diff
@@ -174,6 +178,10 @@ func (r *RunTypeComponentsStub) VmContainerShardFactoryCreator() factoryVm.VmCon
 	return r.VmContainerShardFactory
 }
 
+func (r *RunTypeComponentsStub) AccountsCreator() state.AccountFactory {
```
Comment (sasurobert) [inline testscommon/mainFactoryMocks/runTypeComponentStub.go:181]: comment.

File: factory/runType/sovereignRunTypeComponents.go [hunk 4]
```diff
@@ -18,21 +19,24 @@ import (
...
 	*runTypeComponentsFactory
+	cfg config.SovereignConfig
 }
 
 // NewSovereignRunTypeComponentsFactory will return a new instance of runTypeComponentsFactory
-func NewSovereignRunTypeComponentsFactory(fact *runTypeComponentsFactory) (*sovereignRunTypeComponentsFactory, error) {
+func NewSovereignRunTypeComponentsFactory(fact *runTypeComponentsFactory, cfg config.SovereignConfig) (*sovereignRunTypeComponentsFactory, error) {
 	if check.IfNil(fact) {
 		return nil, errors.ErrNilRunTypeComponentsFactory
 	}
```
Comment (axenteoctavian) [inline factory/runType/sovereignRunTypeComponents.go:36]: or maybe test `NativeESDT` here to be covered in unit tests

File: state/accounts/esdtAsBalance.go [hunk 5]
```diff
@@ -107,6 +103,15 @@ func createEmptyESDT() *esdt.ESDigitalToken {
 	return &esdt.ESDigitalToken{Value: big.NewInt(0), Type: uint32(core.Fungible)}
 }
 
+func fillMandatoryFields(esdtData *esdt.ESDigitalToken) {
+	// make extra sure we have these fields set
+	if esdtData.Value == nil {
+		esdtData.Value = big.NewInt(0)
+	}
+
+	esdtData.Type = uint32(core.Fungible)
```
Comment (axenteoctavian) [inline state/accounts/esdtAsBalance.go:112]: This is not the right implementation to enforce the type here. Starting with `RunTypeComponents` the type should be validated based on `NativeESDT` from config. Getting the balance from account should not change token type, we just use add/sub an amount, the type should always be valid.

File: testscommon/mainFactoryMocks/runTypeComponentStub.go (hunk 3 above)
Comment (mariusmihaic) [reply testscommon/mainFactoryMocks/runTypeComponentStub.go:181]: Right, added comment

File: state/accounts/esdtAsBalance.go (hunk 5 above)
Comment (mariusmihaic) [reply state/accounts/esdtAsBalance.go:112]: We just load a token name from config, it has no properties.
Also, the native token should be and only makes sense to be `Fungible`

File: factory/runType/sovereignRunTypeComponents.go (hunk 4 above)
Comment (mariusmihaic) [reply factory/runType/sovereignRunTypeComponents.go:36]: Will add a task for this to be done whenever we also have token prefix.

**Review by sasurobert — APPROVED** [review]

**Review by axenteoctavian — APPROVED** [review]
sill TODO: Check NativeESDT validity somewhere

**Pull request merged**